lib/matplotlib/streamplot.py

```diff
@@ -206,25 +214,30 @@ def streamplot(axes, x, y, u, v, density=1, linewidth=None, color=None,
             points = np.transpose([tx, ty])
             streamlines.append(points)
 
-        # Add arrows halfway along each trajectory.
+        # Distance along streamline
         s = np.cumsum(np.hypot(np.diff(tx), np.diff(ty)))
-        n = np.searchsorted(s, s[-1] / 2.)
-        arrow_tail = (tx[n], ty[n])
-        arrow_head = (np.mean(tx[n:n + 2]), np.mean(ty[n:n + 2]))
-
-        if isinstance(linewidth, np.ndarray):
-            line_widths = interpgrid(linewidth, tgx, tgy)[:-1]
-            line_kw['linewidth'].extend(line_widths)
-            arrow_kw['linewidth'] = line_widths[n]
-
         if use_multicolor_lines:
             color_values = interpgrid(color, tgx, tgy)[:-1]
             line_colors.append(color_values)
-            arrow_kw['color'] = cmap(norm(color_values[n]))
 
-        p = patches.FancyArrowPatch(
-            arrow_tail, arrow_head, transform=transform, **arrow_kw)
-        arrows.append(p)
+        # Add arrows along each trajectory.
+        for x in range(1, n_arrows+1):
+            # Get index of distance along streamline to place arrow
+            idx = np.searchsorted(s, s[-1] * (x/(n_arrows+1)))
+            arrow_tail = (tx[idx], ty[idx])
+            arrow_head = (np.mean(tx[idx:idx + 2]), np.mean(ty[idx:idx + 2]))
+
+            if isinstance(linewidth, np.ndarray):
+                line_widths = interpgrid(linewidth, tgx, tgy)[:-1]
+                line_kw['linewidth'].extend(line_widths)
```

Comment:
This can't be right? It'll extend `line_kw['linewidth']` for each arrow, but it's passed to the `LineCollection`.

Reply:
Adding `num_arrows` to the Varying line width plot in the example produces:
![Screenshot From 2024-11-15 22-49-40](https://github.com/user-attachments/assets/a6627e30-418d-4db8-9eac-2571bcd16da2)
but the correct result should be:
![Screenshot From 2024-11-15 22-49-51](https://github.com/user-attachments/assets/7f4d5160-0235-409c-ba8f-8fd6e0f62c4e)

This should be fixed now.